components/automate-ui/src/app/modules/infra-proxy/create-chef-server-modal/create-chef-server-modal.component.spec.ts

```diff
@@ -35,18 +39,92 @@ describe('CreateChefServerModalComponent', () => {
   beforeEach(() => {
     fixture = TestBed.createComponent(CreateChefServerModalComponent);
     component = fixture.componentInstance;
+    // This form must mimic the createForm including Validators
     component.createForm = new FormBuilder().group({
-      id: ['', null],
-      name: ['', null],
-      description: ['', null],
-      fqdn: ['', null],
-      ip_address: ['', null]
+      name: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]],
+      id: ['',
+        [Validators.required, Validators.pattern(Regex.patterns.ID), Validators.maxLength(64)]],
+      fqdn: ['', [Validators.required,
+      Validators.pattern(Regex.patterns.NON_BLANK),
+      Validators.pattern(Regex.patterns.VALID_FQDN)
+      ]],
+      ip_address: ['', [Validators.required,
+      Validators.pattern(Regex.patterns.NON_BLANK),
+      Validators.pattern(Regex.patterns.VALID_IP_ADDRESS)
+      ]]
     });
     component.conflictErrorEvent = new EventEmitter();
+    createForm = component.createForm;
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  describe('form validity', () => {
+    describe('the form should be invalid', () => {
+      it('when all inputs are empty', () => {
+        expect(createForm.valid).toBeFalsy();
+      });
+
+      it('when only 3 of the 4 inputs are complete', () => {
+        createForm.controls['name'].setValue('test');
+        createForm.controls['id'].setValue('test');
+        createForm.controls['fqdn'].setValue('test.net');
+
+        errors = createForm.controls['ip_address'].errors || {};
+
+        expect(createForm.valid).toBeFalsy();
+        expect(errors['required']).toBeTruthy();
+      });
+
+      it('when the ip_address in invalid', () => {
+        createForm.controls['name'].setValue('test');
+        createForm.controls['id'].setValue('test');
+        createForm.controls['fqdn'].setValue('chef.internal');
+
+        createForm.controls['ip_address'].setValue('1.2234.3.4');
+        errors = createForm.controls['ip_address'].errors || {};
+
+        expect(createForm.valid).toBeFalsy();
+        expect(errors['pattern']).toBeTruthy();
+      });
+
+      it('when the fqdn Top Level Domain is less than 2 characters', () => {
```

Comment:
Right idea, but you actually only have two invalid and one valid variation of FQDN and there are many more that should be tested.

I just did a quick search and found that the FQDN is already being tested here: https://github.com/chef/automate/blob/df87097454ab61c1c0d49a1a62dfe3ba55dc8648/components/automate-ui/src/app/modules/infra-proxy/chef-servers-list/chef-servers-list.component.spec.ts#L199

There are eight variations tested there--better than three, yes, but still not nearly enough. If you just look at the cyclomatic complexity implicit in the regex...
<img width="1086" alt="image" src="https://user-images.githubusercontent.com/6817500/113353642-d8161980-92f2-11eb-8026-02b106d3b015.png">
...you should appreciate there are more than 8 test cases lurking there. There are, in fact, a couple hundred just at a quick glance. 
But it is tricky to get the right regex. Consider this exercise [In search of the perfect URL validation regex](https://mathiasbynens.be/demo/url-regex) by Mathias Bynens. Among all of those suggested regexes NONE of them got a perfect score. Only one (@diegoperini) even got a great score. And though there are a lot of good tests there that we could use, it does not have one for the case at hand in this PR!

So my suggestion. Either:
(a) use one of the regexes from Bynens chart -- @diegoperini while almost perfect, is too long (thus too unwieldy) for our needs.  I would go with a shorter one that has none (or very few false negatives); less concerned about false positives.
(b) shorten our existing one (again, because it is just not that critical). Possibly like this:
```
^(https?:\/\/)?([a-zA-Z0-9_-]+)*\.[a-zA-Z0-9_-]{2,25}(:[0-9]{1,5})?(\/.*)?$
```
which renders like this, so a lot less variations to test:
<img width="948" alt="image" src="https://user-images.githubusercontent.com/6817500/113355986-748deb00-92f6-11eb-8b52-9bcaf9d7b01b.png">

Then I would suggest the `using` construct for testing a whole bunch of valid URLs and another `using` for testing a whole bunch of invalid ones. Each datum should have a test url and a name/description. 
That will make a much more concise list of tests even than what is in chef-servers-list.component.spec.ts.

For the focus of this PR, valid ones should include:
- http://foo.com.a, one-char suffix
- http://foo.com.ab, two-char suffix
- http://foo.com.1234567890123456789012345, 25-char suffix

Invalid ones should have
- http://foo.com.12345678901234567890123456, 26-char suffix
- http://foo.com., 0-char suffix

Other things to test, to give you a starting point:
- two adjacent dots (fail)
- with or without http:// (succeed)
- with or without https:// (succeed)
- with hyphens (succeed)
- with underscores (succeed)
- with digits (succeed)
- with ports like 0, 23, 19999 (succeed)
- with ports like a, 123456 (fail)
- with final colon but no port chars following (fail)